recommender/engines/rl/ml_components/sarses_generator.py

```diff
@@ -152,6 +162,66 @@ def _get_next_recommendation(recommendation):
     return next_recommendation
 
 
+def find_recs_based_on_uas(ua: UserAction) -> Union[Recommendation, None]:
+    """Find user action's source recommendation."""
+
+    while True:
+        prev_ua = ua
+        svid = prev_ua.source.visit_id
+        recommendation = Recommendation.objects(visit_id=svid).first()
+        if recommendation is not None:
+            return recommendation
+        ua = UserAction.objects(target__visit_id=svid).first()
+        if ua is None:
+            return None
+
+
+def get_recs_for_update(verbose=True) -> List[Recommendation]:
+    """Get recommendations that should be used to select SARSes for update"""
+
+    # Get recommendations that are not processed so far
+    # It takes care about recommendations that aren't root for any user actions
+    if verbose:
+        logger.info("Processing not processed recommendations...")
+    recs_for_update_direct = set(Recommendation.objects(processed__in=[False, None]))
+    Recommendation.objects(processed__in=[False, None]).update(processed=True)
```

Comment:
unncessary double retrieval from DB, should be:
` recs_for_update_direct = set(Recommendation.objects(processed__in=[False, None]).update(processed=True))`

Reply:
or something more readable

Reply:
But this code:
`recs_for_update_direct = set(Recommendation.objects(processed__in=[False, None]).update(processed=True))`
returns set(<number_of_updated_objects>)

To be honest I don't have an idea how to find, update and return those objects using one query.
Do you have any idea?

Reply:
This code will be changed anyway in the near future regarding to: https://github.com/cyfronet-fid/recommender-system/issues/318

Reply:
you can still extract `Recommendation.objects(processed__in=[False, None])` to a variable and then use it accordingly, w/o double DB retrieval